server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/issue/ComponentIssuesLoader.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.computation.task.projectanalysis.issue;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.sonar.api.rule.RuleKey;
+import org.sonar.api.rule.RuleStatus;
+import org.sonar.core.issue.DefaultIssue;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.issue.IssueMapper;
+import org.sonar.server.computation.task.projectanalysis.qualityprofile.ActiveRulesHolder;
+
+public class ComponentIssuesLoader {
+  private final DbClient dbClient;
+  private final RuleRepository ruleRepository;
+  private final ActiveRulesHolder activeRulesHolder;
+
+  public ComponentIssuesLoader(DbClient dbClient, RuleRepository ruleRepository, ActiveRulesHolder activeRulesHolder) {
+    this.activeRulesHolder = activeRulesHolder;
+    this.dbClient = dbClient;
+    this.ruleRepository = ruleRepository;
+  }
+
+  public List<DefaultIssue> loadForComponentUuid(String componentUuid) {
+    try (DbSession dbSession = dbClient.openSession(false)) {
+      List<DefaultIssue> result = new ArrayList<>();
+      dbSession.getMapper(IssueMapper.class).scrollNonClosedByComponentUuid(componentUuid, resultContext -> {
+        DefaultIssue issue = (resultContext.getResultObject()).toDefaultIssue();
+
+        // TODO this field should be set outside this class
+        if (!isActive(issue.ruleKey()) || ruleRepository.getByKey(issue.ruleKey()).getStatus() == RuleStatus.REMOVED) {
+          issue.setOnDisabledRule(true);
+          // TODO to be improved, why setOnDisabledRule(true) is not enough ?
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135)